one_fm/api/doc_events.py

```diff
@@ -244,3 +244,67 @@ def update_training_event_data(doc, method):
 				'training': doc.event_name,
 			})
 			doc_esm.save()
+
+
+def after_insert_of_payroll_entry(doc, method):
+	if doc.employees:

```

Comment:
Did you test this, when payroll_entry is created, employees is not created, employees is added later. This approach is wrong.

Reply:
All you need do is use the payroll_entry start and end date, check HR and Payroll settings for payroll cycle and project within the start and end date.
Then fetch employees in the projects fetched, use the start and end date to get site allowance

Reply:
Oh, ok. When i tested it, i added employees on creation, i will make changes